OVERVIEW.md

```diff
@@ -0,0 +1,265 @@
+    
+    in the matrix case, instead of recasting tangent vectors to 1D by default, 
+    interfacing with some properly einsum-vectorized methods such as 
+    `bilinear: (tensor, vector, vector) -> scalar`, and 
+    `add`, `span`... defined in each of the two parent classes 
+    could be more convenient.
+
++ in the `EmbeddedManifold` class, the Levi-Civita connection 
+can be numerically integrated
+by orthogonal projections onto the the tangent space. 
+This means that by default, `exp` and `log` may derive from `to_tangent` 
+_if not overriden by closed formulas_.
+
+    The `EmbdeddedManifold` class could then be used to represent 
+    manifolds defined as level surfaces `{ f = cst }` 
+    of smooth functions and their intersections, using autograd :scream:.  
+
+    ODE solving code should be kept separate in some
+    `integrators.py` file by an import statement though.
+    Hence the `EmbeddedManifold` class file would only define its objects interfaces
+    (of which already defined manifolds would only inherit the metric tensor), 
+    and this file could as well be imported to solve user-defined ODEs, 
+    on the sphere for instance. 
+
++ As discussed with @ninamiolane
+ and concerning Yann's polydisks, Alice's discretized curves 
+and @nguigs's landmarks spaces: __product spaces__.  
+
+    A clean `Manifold` class model being figured, many canonical methods could 
+    be inherited by a `ProductManifold` child class. 
+    I think this is also related to the first point 
+    (matrices being 2D and vectorization) 
+    as products yield new object shapes and may not always 
+    be represented by rectangular arrays, 
+    thus raising the concern for a common interface reference. 
+
+### Classes
+
+Here the inheritance diagram with the different class
+names is somehow still unclear. A picture attempt:
+```python
+    Connection <---- LeviCivita                             
+                         .                                  
+                         .                                  
+                                                           
+                       Metric                    Vector                
+                         .                         ^     
+                         .                         '       
+                                        .--- EmbeddedManifold
+                     RiemannianMfd <---(                        
+                                        `--- ProductRiemannianMfd
+                                                   :        
+                                                   v         
+                                                Product            
+```         
+
+I think the picture might get simpler 
+if metrics, connections, etc. were thought of as
+auxiliary objects serving to provide/override a manifold's 
+'riemannian' methods. Something like:
+```python
+manifold.use_metric(metric)   
```

Comment:
yup, and this interfacing could be done quite seemlessly with little influence on the code

Reply:
Should the metric inherit its LeviCivita connection or have it as an attribute? For non closed form geodesics, metric.exp() should call connection.exp() which so far calls connection.christoffels. For most manifolds christoffel symbols are not implemented, but a generic method in LeviCivitaConnection can compute them from the metric_matrix. When we have a closed-form expression (e.g. hypersphere in spherical coordinates) where should we put them? See what I did so far in test_connection, I think this line should not be necessary:
https://github.com/geomstats/geomstats/blob/8034dc918dace91b71ca2328963e130bc7cced89/tests/test_connection.py#L81

Reply:
This raises a good question and illustrates a potential issue with having methods `exp` and `log` as attributes of the metric. 

Most implemented manifolds could be provided  a connection with a closed form. 

Then who should `exp` and `log` and `parallel_transport` belong to? the metric or the connection?

Reply:
That's part of the reason why I find the present `manifold.metric.exp` etc. not so good... 

I'd find it more intuitive if we could have the metric litterally "equip" the manifold with such methods, something like:
```python
>>> tens_b, vec_b, point_b = manifold.parallel_transport(tens_a, vec_a, point_a)
 
Error: "manifold" has no metric nor connection, hence no parallel strucure.
```
Something that could be solved by either calling `manifold.use_connection(...)` or `manifold.use_metric(...)`, filling out associated methods accordingly

Reply:
More intuitive and straightforward for the user as well

Reply:
and note that it's not clear that connections would always derive from a metric, for instance on Lie groups you have the canonical [Maurer-Cartan](https://en.wikipedia.org/wiki/Maurer%E2%80%93Cartan_form) one form but you do not always have an invariant _non-degenerate_ bilinear form... 

The Lie group exponential derives from this connection in the sense that left-invariant vector fields on 'G' are measured constant in 'g'. You do not need to introduce this connection to define the exponential though!

 => a few lines away from parallel transport on Lie groups ;)

Reply:
Having RiemannianMetric (identified with its LeviCivita connection) inherits from Connection would solve the ambiguity there, right?

Nice code snippet, I like both manifold.metric.exp and the .use_connection() / .use_metric() and manifold.exp ideas actually!

Reply:
> That's part of the reason why I find the present `manifold.metric.exp` etc. not so good...
> 
> I'd find it more intuitive if we could have the metric litterally "equip" the manifold with such methods, something like:
> 
> ```python
> >>> tens_b, vec_b, point_b = manifold.parallel_transport(tens_a, vec_a, point_a)
>  
> Error: "manifold" has no metric nor connection, hence no parallel strucure.
> ```
> 
> Something that could be solved by either calling `manifold.use_connection(...)` or `manifold.use_metric(...)`, filling out associated methods accordingly

it would be more user-friendly indeed, but how would you implement the `manifold.use_metric` method?